ezb_api/ezb_api.py

```diff
@@ -0,0 +1,342 @@
+    if validate_curr_requirements(source, target):
+        currency_pair_exchange_rate_url = _EXCHANGE_RATE_URL.format(source, target)
+        ezb_request = get_response_from_ezb(currency_pair_exchange_rate_url)
+        if check_request_response(ezb_request):
+            curr_exchange_rate_pdf = get_pdf_from_ezb_xml(ezb_request)
+            return curr_exchange_rate_pdf
+        else:
+            raise ValueError(
+                f"REST API retrieves for {currency_pair_exchange_rate_url} following code: {ezb_request.status_code}")
+
+    else:
+        raise ValueError("Currency does not match the requirements.")
+
+
+def check_identifier(identifier: str):
+    """Check validity of the identifier.
+
+    Args:
+        identifier (str): the identifier given as string
+
+    Returns:
+        A boolean, it is false if identifier does not statisfy the requirements and true if it matches the requirements.
+    """
+    validation_result = False
+    identifier = str(identifier)
+    if re.search("^[A-Z0-9.\_]*$", identifier):
+        identifier_elements = identifier.split('.')
+        if len(identifier_elements) > _INDEX_SOURCE_CURRENCY_IDENTIFY:
+            if validate_curr(identifier_elements[_INDEX_SOURCE_CURRENCY_IDENTIFY]):
+                return True
+    return validation_result
+
+
+def get_raw_data(identifier: str) -> pd.DataFrame:
+    """Creates pandas dataframe from the retrieved data, based on the identifier.
+
+    Args:
+        identifier (str): the identifier given as string
+
+    Returns:
+        A pandas dataframe with retrieved data from the api.
+
+    Raises:
+        ValueError: An error occurred if identifier is not matching the requirements.
+    """
+    if check_identifier(identifier):
+        identifier_url = _IDENTIFIER_URL.format(identifier)
+        ezb_request = get_response_from_ezb(identifier_url)
+        ezb_data_pdf = get_pdf_from_ezb_xml(ezb_request)
+        return ezb_data_pdf
+    else:
+        raise ValueError("Identifier is not matching the requirements")
+
+
+def extract_currency_from_identifier(identifier: str):
+    """Extracts currency from the api.
+
+    Args:
+        identifier (str): the identifier given as string
+
+    Returns:
+        Return None if no currency exist in identifier, otherwise the currency will be returned.
+
+    Raises:
+        ValueError: An error occurred if identifier is not matching the requirements
+    """
+    currency = None
+    if check_identifier(identifier):
+        identifier = str(identifier)
+        identifier_elements = identifier.split('.')
+        currency = identifier_elements[_INDEX_SOURCE_CURRENCY_IDENTIFY]
+    else:
+        raise ValueError(f"Identifier does not met the requirements: {identifier}")
+    return currency
+
+
+def validate_dataframes_for_multiplication(df_left: pd.DataFrame, df_right: pd.DataFrame, calc_col_name: str = 'OBS_VALUE') -> pd.DataFrame:
+    """Validates if multiplikation between two dataframes is possible.

```

Comment:
Very german spelling of multiplication, makes harder to search for it in the code